### PR TITLE
[Ft #159356728]Add support for periodization

### PR DESCRIPTION
**Pull request by oagutu**
#### What does this PR do?
Add support for macrocycle, mesocycle and microcycle. Macro is an annual plan, meso is phase of training with a duration of between 2-6 weeks and micro is instead only a week.

#### Description of Task to be completed?
 - A user should be able to create a schedule for one of the periodisation cycles, ie. Macrocycle, mesocycle or microcycle. This is achieved by specifying the desired period when adding a new schedule
 - The duration of workouts added to a cycle depends on the cycle type. Eg. a mesocycle will not allow  a user to add workouts with a total duration exceeding 6 weeks. 

#### How should this be manually tested?
 - Once logged in to the app, a user may create a workout. 
 - The user may then create a new schedule specifying desired cycle type eg. mesocycle.
 - Once the schedule is created, the user should add the created workout, setting a duration higher than 6 weeks
 - The user should be alerted of invalid duration as it should not last more than 6 weeks.
#### Any background context you want to provide?
This feature is needed for sports periodization ( powerlifting, weightlifting, strongman, etc ) or at least to create a fitness programs drafted on more than only one week.
Cycles can be divided into three/four segments, especially for Macrocycles, and so it is recommended they do not exceed 30 weeks.

#### What are the relevant pivotal tracker stories?
[#159356728: Add support for macrocycle, mesocycle and microcycle](https://www.pivotaltracker.com/story/show/159356728)

#### Screenshots:
```Add schedule form showing period selection```
screen shot 2018-08-16 at 13 11 48

```Individual schedule view showing cycle type(ie. Macrocycle) ```
screen shot 2018-08-16 at 13 13 31

```Example of error message when adding workouts to schedule with total cycle duration exceeded(1 year for macrocycle)```
screen shot 2018-08-16 at 13 13 21

### CODE REVIEW

**Comment by Nerldy** [comment]
Good job. I like the dropdown feature Makes it easy for a user to set up.

**Comment by ezrqnkemboi** [comment]
I really like the choice made for verifications.

File: wger/manager/views/schedule_step.py [hunk 1]
```diff
@@ -61,12 +62,43 @@ def get_form_class(self):
...
+                # Check if duration exceeds cycle duration
+                if schedule.period == 'Macrocycle' and derived_duration > 52:
+                    raise ValidationError(_('Invalid duration- cycle duration exceeding 1 year'))
+                elif schedule.period == 'Mesocycle' and derived_duration > 6:
+                    raise ValidationError(_('Invalid duration- cycle duration exceeding 6 weeks'))
+                elif schedule.period == 'Microcycle' and derived_duration > 1:
+                    raise ValidationError(_('Invalid duration- cycle duration exceeding 1 week'))
+
+                return duration
+
```
Comment (3V0L) [inline wger/manager/views/schedule_step.py]: Great work on this function, well documented and adhering to coding conventions set by the team.

**Comment by tapiwa03** [comment]
Great work on this function, well documented and adhering to coding conventions set by the team.

**Pull request merged**